wled00/bus_manager.h

```diff
@@ -122,59 +79,76 @@ class Bus {
-      return false;
+    virtual void     setBrightness(uint8_t b)                     { _bri = b; };
+    virtual void     setColorOrder(uint8_t co)                    {}
+    virtual uint32_t getPixelColor(uint16_t pix) const            { return 0; }
+    virtual uint8_t  getPins(uint8_t* pinArray = nullptr) const   { return 0; }
+    virtual uint16_t getLength(void) const                        { return isOk() ? _len : 0; }
+    virtual uint8_t  getColorOrder(void) const                    { return COL_ORDER_RGB; }
+    virtual uint8_t  skippedLeds(void) const                      { return 0; }
+    virtual uint16_t getFrequency(void) const                     { return 0U; }
+    virtual uint16_t getLEDCurrent(void) const                    { return 0; }
+    virtual uint16_t getUsedCurrent(void) const                   { return 0; }
+    virtual uint16_t getMaxCurrent(void) const                    { return 0; }
+
+    inline  bool     hasRGB(void) const                           { return _hasRgb; }
+    inline  bool     hasWhite(void) const                         { return _hasWhite; }
+    inline  bool     hasCCT(void) const                           { return _hasCCT; }
+    inline  bool     isDigital(void) const                        { return isDigital(_type); }
+    inline  bool     is2Pin(void) const                           { return is2Pin(_type); }
+    inline  bool     isOnOff(void) const                          { return isOnOff(_type); }
+    inline  bool     isPWM(void) const                            { return isPWM(_type); }
+    inline  bool     isVirtual(void) const                        { return isVirtual(_type); }
+    inline  bool     is16bit(void) const                          { return is16bit(_type); }
+    inline  void     setReversed(bool reversed)                   { _reversed = reversed; }
+    inline  void     setStart(uint16_t start)                     { _start = start; }
+    inline  void     setAutoWhiteMode(uint8_t m)                  { if (m < 5) _autoWhiteMode = m; }
+    inline  uint8_t  getAutoWhiteMode(void) const                 { return _autoWhiteMode; }
+    inline  uint8_t  getNumberOfChannels(void) const              { return hasWhite() + 3*hasRGB() + hasCCT(); }
+    inline  uint16_t getStart(void) const                         { return _start; }
+    inline  uint8_t  getType(void) const                          { return _type; }
+    inline  bool     isOk(void) const                             { return _valid; }
+    inline  bool     isReversed(void) const                       { return _reversed; }
+    inline  bool     isOffRefreshRequired(void) const             { return _needsRefresh; }
+    inline  bool     containsPixel(uint16_t pix) const            { return pix >= _start && pix < _start + _len; }
+
+    static std::vector<LEDType> getLEDTypes(void)                 { return {{TYPE_NONE, "", PSTR("None")}}; }
```

Comment:
Do we need this?   It's not called anywhere - maybe a comment indicating that this is a mandatory call on all Bus subclasses would be sufficient.

Reply:
No, we don't. Added just in case someone might convert that into virtual in the future.
Does not seem likely, though. Can be made `inline` and it may not end up in the binary. Or can be commented out.

Reply:
There's no such thing as "static virtual", though.  To call a virtual function, you need an object to look up the function pointer from, so they can't be static.  For now I'd suggest commenting it out.